### ISSUE TITLE
Channel name autocomplete API endpoint returns errors when name contains pipe characters

### ISSUE THREAD

**Issue by WGH-** [mattermost/mattermost-server#8962]
#### Summary
Channel name autocomplete API endpoint returns errors when name contains pipe characters.

#### Steps to reproduce
Occurs on Mattermost 5.0 and earlier versions (at least since 4.7.0, likely much earlier, too).

In almost any input field (e.g. message entry, channel header entry), enter `~` followed by `|`.
This will result in request to the URL like this `/api/v4/teams/xxxxxxxx/channels/autocomplete?name=%7C`

#### Observed behavior (that appears unintentional) 
This will result in the following JSON:
```json
{
  "id": "store.sql_channel.search.app_error",
  "message": "We encountered an error searching channels",
  "detailed_error": "",
  "request_id": "iwimmowym7fcfyhykjtyufoipy",
  "status_code": 500
}
```
The following error will logged on the server:
```json
{
  "level": "error",
  "ts": 1529365231.5370913,
  "caller": "web/handlers.go:140",
  "msg": "We encountered an error searching channels",
  "path": "/api/v4/teams/oqb1ed9r8j8ofezeg94nq1o9nw/channels/autocomplete",
  "request_id": "iwimmowym7fcfyhykjtyufoipy",
  "ip_addr": "zzz.zzz.zzz.zzz",
  "user_id": "xxxxxx",
  "method": "GET",
  "err_where": "SqlChannelStore.AutocompleteInTeam",
  "http_code": 500,
  "err_details": "term=|, , pq: syntax error in tsquery: \"|:*\""
}
```

It wouldn't be much of a problem if autocomplete dropdown didn't get stuck in loading state. Pressing "Enter" in this state will result in all text after (including) the tilde character getting deleted. This is mildly annoying.

#### P.S.

Note that despite SQL errors, this is not a SQL injection, and is not a security bug. "Injection" occurs inside PostgreSQL full-text query minilanguage, where it's unlikely to do any harm. Basically, what happens is:
```
postgres=# SELECT to_tsquery('english', '|:*');
ERROR:  syntax error in tsquery: "|:*"
```

**Comment by amyblais** [comment]
Thanks @WGH-! I created a Jira ticket for this for our engineers to look at: https://mattermost.atlassian.net/browse/MM-10972.

**Comment by amyblais** [comment]
Hi @WGH-! Letting you know that https://mattermost.atlassian.net/browse/MM-10972 has been submitted and the fix is available in 5.1.

5.1 will be released in July 16th.

I'll close this Issue for now since the fix is available soon. If you encounter the bug after upgrading to 5.1, please reopen this Issue.
